### ISSUE TITLE
Stuck at stage 3

### ISSUE THREAD

**Issue by diabeticp** [jwoglom/controlX2#6]
Hi! I have installed latest version on some of my phones. On one of them, I'm stuck at stage 3 "Initial pump metadata received". I've also tried pairing with code through t:connect app, but no luck.

Phone: Samsung Galaxy M52 5G 
Android version: Android 13 (One UI 5.0)
Pump version: 7.3.1
![image](https://user-images.githubusercontent.com/81512138/221523775-90650120-69df-438a-9385-8c052579142d.jpeg)

**Comment by jwoglom** [comment]
I've run into this as well. I'm not completely sure why it happens.

Have you been able to eventually get this to work? I usually need to toggle bluetooth and force-close, and be patient and it eventually connects.

**Comment by diabeticp** [comment]
No, I haven't. I've tried for several hours with force-stopping app, rebooting device, cleaning cache and data, Bluetooth on and off (phone and pump), didn't work at all. 
Idk, maybe it's because of Samsung optimisation (One UI). If I remember correctly, even other diabetic apps don't work so well on Samsung phones (like xDrip+, AndroidAPS, etc.).

**Comment by BenoitAnastay** [comment]
I have the same issue on a european pump and a Pixel 6, it wasn't working with t:connect app so I guessed it was the region that cause the issue, I should run it with a IDE to see if there is an error
